### PR TITLE
[REVIEW] Change  concat to use new libgdf method

### PR DESCRIPTION
**Pull request by dantegd**
Changes to `_gdf.py`, `buffer.py`, `column.py` and `series.py` to call new libgdf concat in PR https://github.com/gpuopenanalytics/libgdf/pull/152

### CODE REVIEW

File: pygdf/_gdf.py [hunk 1]
```diff
@@ -117,19 +117,23 @@ def apply_mask_and(col, mask, out):
...
-        np.float64: libgdf.GDF_FLOAT64,
-        np.float32: libgdf.GDF_FLOAT32,
-        np.int64:   libgdf.GDF_INT64,
-        np.int32:   libgdf.GDF_INT32,
-        np.int16:   libgdf.GDF_INT16,
-        np.int8:    libgdf.GDF_INT8,
-        np.bool_:   libgdf.GDF_INT8,
-        np.datetime64: libgdf.GDF_DATE64,
-    }[np.dtype(dtype).type]
+    if str(dtype) == 'category':
```
Comment (kkraus14) [inline pygdf/_gdf.py]: Fix incoming for this to use `pd.core.common.is_categorical_dtype` instead

File: pygdf/_gdf.py [hunk 2]
```diff
@@ -376,6 +380,12 @@ def hash_partition(input_columns, key_indices, nparts, output_columns):
     return offsets
 
 
+def column_concat(cols_to_concat, output_col):
```
[the remark this answers is not in the document]
Comment (kkraus14) [reply pygdf/_gdf.py]: This is an internal method that shouldn't be used directly by end users so I don't think a docstring is necessary.

[the remark this answers is not in the document]
Comment (scopatz) [reply pygdf/_gdf.py]: Then it seems like it should have an leading underscore in the name, to denote that, eg. `_column_concat()`

**Comment by harrism** [comment]
@dantegd Can you update this PR?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by harrism** [comment]
add to whitelist

File: python/cudf/_gdf.py [hunk 3]
```diff
@@ -131,6 +131,11 @@ def apply_mask_and(col, mask, out):
 def np_to_gdf_dtype(dtype):
     """Util to convert numpy dtype to gdf dtype.
     """
+
+    if pd.core.common.is_categorical_dtype(dtype):
```
Comment (kkraus14) [inline python/cudf/_gdf.py]: This is deprecated and should be removed apparently. Will use the implementation below this :).

**Comment by kkraus14** [comment]
All green, merging! Thanks @dantegd!

**Pull request merged**